frontend/packages/kubevirt-plugin/src/components/vms/vm-users.tsx

```diff
@@ -0,0 +1,152 @@
+import { VMStatusBundle } from '../../statuses/vm/types';
+import { VMIKind } from '../../types';
+
+const guestAgentURL = (name: string, namespace: string) =>
+  `apis/subresources.kubevirt.io/v1alpha3/namespaces/${namespace}/virtualmachineinstances/${name}/guestosinfo`;
+
+const tableColumnClasses = [
+  classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'col-sm-4'),
+  classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'col-sm-4'),
+  classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'col-sm-4'),
+  classNames('col-lg-3', 'col-md-3', 'hidden-sm', 'hidden-xs'),
+];
+
+const UsersTableHeader = () => {
+  return [
+    {
+      title: 'User Name',
+      sortField: 'metadata.userName',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[0] },
+    },
+    {
+      title: 'Domain',
+      sortField: 'metadata.domain',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[1] },
+    },
+    {
+      title: 'Login time',
+      sortField: 'metadata.loginTime',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[2] },
+    },
+    {
+      title: 'Elapsed logged in time',
+      props: { className: tableColumnClasses[3] },
+    },
+  ];
+};
+UsersTableHeader.displayName = 'UsersTableHeader';
+
+const UsersTableRow = ({ obj: user, index, key, style }) => {
+  return (
+    <TableRow id={user?.metadata?.uid} index={index} trKey={key} style={style}>
+      <TableData className={tableColumnClasses[0]}>{user?.metadata?.userName}</TableData>
+      <TableData className={classNames(tableColumnClasses[1], 'co-break-word')}>
+        {user?.metadata?.domain}
+      </TableData>
+      <TableData className={tableColumnClasses[2]}>
+        <Timestamp timestamp={new Date(user?.metadata?.loginTime).toUTCString()} />
+      </TableData>
+      <TableData className={tableColumnClasses[3]}>{fromNow(user?.metadata?.loginTime)}</TableData>
+    </TableRow>
+  );
+};
+
+export const VMUsersListWrapper: React.FC<VMUsersListProps> = ({
+  watchURLAction,
+  stopWatchURLAction,
+  urlResults,
+  vmi,
+  vmStatusBundle,
+}) => {
+  if (vmStatusBundle.status !== VMStatus.RUNNING) {
+    return <div className="text-center">{VIRTUAL_MACHINE_IS_NOT_RUNNING}</div>;
+  }
+
+  if (!isGuestAgentInstalled(vmi)) {
+    return <StatusItem Icon={BlueInfoCircleIcon} message={NO_GUEST_AGENT_MESSAGE} />;
+  }
+
+  const name = getName(vmi);
+  const namespace = getNamespace(vmi);
+  const url = guestAgentURL(name, namespace);
+
+  const usersResult = urlResults.getIn([url, 'data']);
+  const usersResultError = urlResults.getIn([url, 'loadError']);
+
+  const data =
+    usersResult &&
+    usersResult?.userList &&
+    usersResult?.userList.map((user, uid) => ({
+      metadata: {
+        uid,
+        userName: user?.userName,
+        domain: user?.domain,
+        loginTime: user?.loginTime && user.loginTime * 1000,
+      },
+    }));
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  React.useEffect(() => {
```

Comment:
this is a coditional usage of hook. It needs to be moved on top before all if/returns

Reply:
done